### PR TITLE
OpenSearchSink - Enhance logs to include index name and last exception information

### PR DESCRIPTION
**Pull request by kkondaka**
### Description
OpenSearchSink - 
   Enhance logs to include index name where applicable
   and add last exception information when giving up after retrying
 
### Issues Resolved
Resolves #[Issue number to be closed when this PR is merged]
 
### Check List
- [ ] New functionality includes testing.
- [ ] New functionality has a documentation issue. Please link to it in this PR.
  - [ ] New functionality has javadoc added
- [ X] Commits are signed with a real name per the DCO

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/data-prepper/blob/main/CONTRIBUTING.md).

### CODE REVIEW

**Review by graytaylor0 — APPROVED** [review]

**Comment by graytaylor0** [comment]
Build and DCO checks are failing

File: data-prepper-plugins/opensearch/src/main/java/org/opensearch/dataprepper/plugins/sink/opensearch/BulkRetryStrategy.java [hunk 1]
```diff
@@ -209,7 +209,7 @@ public void execute(final AccumulatingBulkRequest bulkRequest) throws Interrupte
                 request = operationResponse.getBulkRequest();
                 exceptionMessage = operationResponse.getExceptionMessage();
                 if (delayMillis < 0) {
-                    RuntimeException e = new RuntimeException(String.format("Index : %s, Number of retries reached the limit of max retries (configured value %d. Last exception message: %s)", maxRetries, exceptionMessage));
```
Comment (graytaylor0) [inline data-prepper-plugins/opensearch/src/main/java/org/opensearch/dataprepper/plugins/sink/opensearch/BulkRetryStrategy.java]: Why get rid of the index name here?

**Review by graytaylor0 — APPROVED** [review]

File: data-prepper-plugins/opensearch/src/main/java/org/opensearch/dataprepper/plugins/sink/opensearch/BulkRetryStrategy.java (hunk 1 above)
Comment (kkondaka) [reply data-prepper-plugins/opensearch/src/main/java/org/opensearch/dataprepper/plugins/sink/opensearch/BulkRetryStrategy.java]: @graytaylor0 I did not get rid of it. It is not available. I have mistakenly put it there in the previous revision.

Comment (kkondaka) [reply data-prepper-plugins/opensearch/src/main/java/org/opensearch/dataprepper/plugins/sink/opensearch/BulkRetryStrategy.java]: Index is available at each operation level. Not at bulk operation level.

**Review by KarstenSchnitter — COMMENTED** [review]
Is it possible for the exception message to contain sensitive information? In that case, I think the marker should be set in the log statements.

**Comment by kkondaka** [comment]
@KarstenSchnitter most of the changes are about adding index name. There is one place where exception message is printed. But this is no different from what we already do.

**Review by graytaylor0 — APPROVED** [review]

**Review by dlvenable — APPROVED** [review]

**Review by KarstenSchnitter — APPROVED** [review]

**Pull request merged**